nanome/_internal/_network/_data.py

```diff
@@ -28,20 +30,18 @@ def received_data(self, data):
 
     def expand_data(self, size):
         self._buffered_bytes += size
-        if (self._buffered_computed + self._buffered_bytes > self._end):
-            self._expand_data(size)
+        while (self._buffered_computed + self._buffered_bytes > self._end):
```

Comment:
This will probably be slower in the case we try to write_bytes a huge chunk of data. But it is faster for our common operations (allocating only 1 MB at a time)